modules/imgproc/src/imgwarp.hpp

```diff
@@ -67,17 +68,18 @@ void convertMaps_nninterpolate32f1c16s_SSE41(const float* src1f, const float* sr
 void convertMaps_32f1c16s_SSE41(const float* src1f, const float* src2f, short* dst1, ushort* dst2, int width);
 void convertMaps_32f2c16s_SSE41(const float* src1f, short* dst1, ushort* dst2, int width);
 void WarpAffineInvoker_Blockline_SSE41(int *adelta, int *bdelta, short* xy, int X0, int Y0, int bw);
-
-class WarpPerspectiveLine_SSE4
-{
-public:
-    static Ptr<WarpPerspectiveLine_SSE4> getImpl(const double *M);
-    virtual void processNN(const double *M, short* xy, double X0, double Y0, double W0, int bw) = 0;
-    virtual void process(const double *M, short* xy, short* alpha, double X0, double Y0, double W0, int bw) = 0;
-    virtual ~WarpPerspectiveLine_SSE4() {};
-};
 #endif
 }
+
+#if CV_TRY_SSE4_1
+void WarpPerspectiveLine_ProcessNN_SSE41(const double *M, short* xy, double X0, double Y0, double W0, int bw);
```

Comment:
I think it would be better to retain SSE4_1 optimizations inside `opt_SSE4_1` namespace to avoid possible name conflicts

Reply:
Restored